js/wallet.js

```diff
@@ -40,10 +104,126 @@ class JS extends Wallet {
     mockery.registerMock('../../../js/lib/request', {
       request: request
     })
+    mockery.registerMock('../../../js/settings', {
+      getSetting: (key) => {
+        switch (key) {
+          case settings.PAYMENTS_MINIMUM_VISITS:
+          {
+            return self.minimumVisits
+          }
+          case settings.PAYMENTS_MINIMUM_VISIT_TIME:
+          {
+            return self.minimumVisitTime
+          }
+          case settings.PAYMENTS_CONTRIBUTION_AMOUNT:
+          {
+            return self.defaultContribution
+          }
+          case settings.PAYMENTS_ENABLED:
+          case settings.PAYMENTS_NOTIFICATION_TRY_PAYMENTS_DISMISSED:
+          {
+            return true
+          }
+        }
+      }
+    })
 
     this.ledger = require('../browser-laptop/app/browser/api/ledger')
+    this.ledgerStatuses = require('../browser-laptop/app/common/constants/ledgerStatuses')
+    this.setStateFile()
+
+    this.resetWalletProperties = function () {
+      self.wallet = 'default-wallet'
+      self.stateKey = 'default-state'
+      self.setStateFile()
+
+      const stateFile = JSON.parse(self.stateFile)
+      const body = Immutable.fromJS(responses['/v2/wallet/balance'][self.wallet])
+
+      self.setState(self.ledger.onWalletProperties(self.state, body))
+      self.setState(self.setInfo(self.state, 'paymentId', stateFile.properties.wallet.paymentId))
+      self.setState(self.ledger.onInitRead(self.state, stateFile))
+    }
+
+    this.callback = sinon.stub(this.ledger, 'callback').callsFake(function (err, result, delayTime) {
```

Comment:
Do we need to assign every stub to a variable? We don't use it anywhere

Reply:
You are correct that we don't use it for now, I was planning on requiring these in after some future refactoring. Maybe this would be better to reorganize in a different PR?

Reply:
sure sounds good